internal/v2/s2av2.go

```diff
@@ -131,7 +132,22 @@ func (c *s2av2TransportCreds) ClientHandshake(ctx context.Context, serverAuthori
 	serverName := removeServerNamePort(serverAuthority)
 	timeoutCtx, cancel := context.WithTimeout(ctx, GetS2ATimeout())
 	defer cancel()
-	s2AStream, err := createStream(timeoutCtx, c.s2av2Address, c.getS2AStream)
+	var err error
+	var s2AStream stream.S2AStream
+	retryer := NewRetryer()
+	for {
+		if err = ctx.Err(); err != nil {
+			break
+		}
+		s2AStream, err = createStream(timeoutCtx, c.s2av2Address, c.getS2AStream)
+		if backoff, shouldRetry := retryer.Retry(err); shouldRetry {
+			if sleepErr := retry.Sleep(ctx, backoff); sleepErr != nil {
```

Comment:
Should we log (maybe VLOG(1) if we're retrying)?

Similarly throughout.

Reply:
done